### PR TITLE
PUPIL-292- feat(pupil-lesson-review-item): create review item and add to storybook

### PR DESCRIPTION
**Pull request by benprotheroe**
create OakLessonReviewItem component and add to strybook

the component is to be used in the pupil view lesson review page

### CODE REVIEW

**Comment by rightsaidjames** [comment]
@benprotheroe Can you add a how to test section, screenshots etc.?

**Review by messmike — APPROVED** [review]

File: src/components/integrated/OakLessonReviewItem/OakPupilLessonReviewItem.tsx [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+          {pickSummaryForProgress(props)}
+        </OakBox>
+      </OakFlex>
+      {renderQuestionCounter(props)}
+    </StyledLessonReviewItem>
+  );
+};
+
+function pickSummaryForProgress(props: OakLessonReviewItemProps) {
+  switch (props.completed) {
```
Comment (kimon-satan) [inline src/components/integrated/OakLessonReviewItem/OakPupilLessonReviewItem.tsx:83]: Small code style thing, but while switch is the best option for when there are multiple conditions it doesn't make sense for something as simple as this. Here I would use a simple if and return early. Then follow up with the else condition (although no need to actually use the else keyword)

File: src/components/integrated/OakLessonReviewItem/OakPupilLessonReviewItem.tsx [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    case "starter-quiz":
+      return ["bg-decorative1-very-subdued", "border-decorative1", "mint"];
+    case "video":
+      return ["bg-decorative4-very-subdued", "border-decorative4", "pink"];
+    case "exit-quiz":
+      return ["bg-decorative5-very-subdued", "border-decorative5", "lemon"];
+  }
+}
+
+function pickIconForSection(sectionName: LessonSectionName): OakIconName {
```
Comment (kimon-satan) [inline src/components/integrated/OakLessonReviewItem/OakPupilLessonReviewItem.tsx:145]: Again small style thing - It's nice to separate these things out but a function feels like over kill here. How about using a Map instead ?

File: src/components/integrated/OakLessonReviewItem/OakPupilLessonReviewItem.tsx [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+      return "intro";
+    case "starter-quiz":
+    case "exit-quiz":
+      return "quiz";
+    case "video":
+      return "video";
+  }
+}
+
+function renderQuestionCounter(props: OakLessonReviewItemProps) {
```
Comment (kimon-satan) [inline src/components/integrated/OakLessonReviewItem/OakPupilLessonReviewItem.tsx:157]: One more code style thing. I prefer arrow functions whenever possible. I meant to enforce this in the linter but never got round to it. I don't want these to creep in too much.

**Review by kimon-satan — CHANGES_REQUESTED** [review]
I made a few code style suggestions but otherwise all good.

File: src/components/integrated/OakLessonReviewItem/OakPupilLessonReviewItem.tsx [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+  OakFlex,
+  OakIcon,
+  OakIconName,
+  OakSpan,
+} from "@/components/base";
+import { OakRoundIcon } from "@/components/ui";
+import { OakCombinedColorToken } from "@/styles";
+
+type LessonSectionName = "intro" | "starter-quiz" | "video" | "exit-quiz";
+type BaseOakLessonReviewItenProps = {
```
Comment (kimon-satan) [inline src/components/integrated/OakLessonReviewItem/OakPupilLessonReviewItem.tsx:9]: Typo

File: src/components/integrated/OakLessonReviewItem/OakPupilLessonReviewItem.tsx [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+const StyledLessonReviewItem = styled(OakFlex)<{ completed: boolean }>`
+  outline: none;
+  text-align: initial;
+`;
+
+export const OakLessonReviewItem = (props: OakLessonReviewItemProps) => {
+  const { completed, lessonSectionName, ...rest } = props;
+  const [completedBackgroundColor, borderColor, iconBackgroundColor] =
+    pickColorsForSection(lessonSectionName);
+  const map = new Map();
```
Comment (kimon-satan) [inline src/components/integrated/OakLessonReviewItem/OakPupilLessonReviewItem.tsx:43]: Can you give this a better name please

File: src/components/integrated/OakLessonReviewItem/OakPupilLessonReviewItem.tsx [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+   * The number of questions in the quiz
+   */
+  numQuestions: number;
+  /**
+   * The number of questions answered correctly
+   */
+  grade: number;
+};
+
+type VideoSectionProps = {
```
Comment (kimon-satan) [inline src/components/integrated/OakLessonReviewItem/OakPupilLessonReviewItem.tsx:24]: These two prop types could be combined

**Review by kimon-satan — APPROVED** [review]
Put a couple more tidy up points but have approved. Could you please do what James asked in his comment and re-request a review from him.

**Pull request merged**